src/twisted/application/twist/test/test_twist.py

```diff
@@ -201,3 +201,102 @@ def run(self):
             )
         )
         self.assertEqual(runners[0].runs, 1)
+
+
+
+class CrippledRunner(object):
+    def __init__(self, **kwargs):
+        self.args = kwargs
+
+    def run(self):
+        pass
+
+
+
+class TwistExitTests(twisted.trial.unittest.TestCase):
+
+    def setUp(self):
+        self.exitWithSignalCalled = False
+
+        def startService(service):
+            pass
+
+        self.patch(MultiService, "startService", startService)
```

Comment:
Actually, changing the plugin called (e.g. to words) would also eliminate the need for that patching as well.  When the web plugin's makeService executes it calls strports.service, which imports the default reactor if one isn't passed in and results in the reactory being dirty when the test exits.

Regardless, I'd prefer something a little more deterministic than relying on the current implementation of whatever plugin is chosen for the test.